### ISSUE TITLE
Add votes to comments

### ISSUE THREAD

**Issue by riggraz** [astuto/astuto#408]
Add the possibility to vote comments.

For implementation purposes, it may be useful to use [Rails polymorphic associations](https://guides.rubyonrails.org/association_basics.html#polymorphic-associations)

**Comment by hitenvidhani** [comment]
Hi @riggraz 
Do we want to add a functionality to up-vote and/or down-vote comments here, similar to stack overflow?
![image](https://github.com/user-attachments/assets/d9195672-cc49-45f4-a718-47bc37f35c7c)

**Comment by riggraz** [comment]
Hi @hitenvidhani

Downvotes are interesting and something I'm considering.

However (1) I'd make them optional (i.e. site administrators can decide whether to allow downvotes or not) and

(2) I have the following doubt: votes are visible to administrators (i.e. they can see name and email of who voted) so maybe users will be worried to downvote a feedback knowing that. But maybe, since votes can only be seen by administrators and not normal users, that's not a problem.

Let me know what you think!

**Comment by hitenvidhani** [comment]
> However (1) I'd make them optional (i.e. site administrators can decide whether to allow downvotes or not) and

I agree with you on this, but I feel that site admins should also be able to decide whether they even want to enable voting. _Enable_/Disable Voting -> Enable/Disable Downvoting

> I have the following doubt: votes are visible to administrators (i.e. they can see name and email of who voted) so maybe users will be worried to downvote a feedback knowing that. But maybe, since votes can only be seen by administrators and not normal users, that's not a problem.

I feel here we can provide an option for the admins to have a flag with which they can decide whether they want their voters' info. This will be then serving both use cases.

What do you think about this approach of giving admin the control of information.

**Comment by riggraz** [comment]
Hi @hitenvidhani, thanks for your suggestions!

> I feel that site admins should also be able to decide whether they even want to enable voting

I agree. Admins should be able to enable/disable upvotes as well as downvotes. Maybe upvotes will be enabled by default and downvotes disabled by default.

> I feel here we can provide an option for the admins to have a flag with which they can decide whether they want their voters' info. This will be then serving both use cases.

I don't know about this. Maybe I'd just put a disclaimer somewhere reminding users that their votes will be visible to admins/mods only.

**Comment by riggraz** [comment]
Moved the discussion on downvotes to #421 !
